config/config_mgmt.py

```diff
@@ -545,7 +544,7 @@ def _mergeItems(it1, it2):
                     pass
                 return
 
-            for it in D1.keys():
+            for it in list(D1.keys()):
```

Comment:
```suggestion
            for it in D1:
```

Reply:
This change was made by the `2to3` tool to retain backward-compatibility with minimal changes. Since sonic-utilities is not a library, there really isn't a need to maintain backward-compatibility. My intention here was only to maintain backward-compatibility long enough to allow for a submodule update and to update sonic-buildimage to build the Python 3 package, otherwise, once this PR merges, the build will be broken until sonic-buildimage is updated. However, it would save some work to eliminate Python 2 support altogether here, but it might result in broken PR builds for a day or so until everything is updated accordingly.

Reply:
It will work. I think you are talking about
```
for it, _ in D1.items():
```

---
In reply to: [499074240](https://github.com/Azure/sonic-utilities/pull/1128#discussion_r499074240) [](ancestors = 499074240)

Reply:
Yes, you're correct. I misread this one. Everything started blending together after all these changes :)

I will go back through and do a cleanup pass in the future.

Reply:
I have opened a PR to clean up [here](https://github.com/Azure/sonic-utilities/pull/1243). However, this instance must iterate over `D1.keys()`, because elements are deleted from the dictionary in the loop. Using `for it in D1:` here results in a crash when the dictionary is altered.